src/main/scala/lms/transformation/distributed_tensor/distributedTensorUnary.scala

```diff
@@ -24,9 +24,27 @@ trait FixedSizeDistributedTensorUnaryTypeLess extends FixedSizeDistributedTensor
     (new TENSOR(Adapter.g.reflectRead("tensor_negate", C(res_tt), C(anno), tensor.x)(tensor.x)).withSrcType(__pos, tensor.et))
   }
 
+  def Invert(tensor: TENSOR, anno: Anno = NAnno)(implicit __pos: SourceContext): TENSOR = {
+    val res_tt = tensor.resultType
+    (new TENSOR(Adapter.g.reflectRead("tensor_invert", C(res_tt), C(anno), tensor.x)(tensor.x)).withSrcType(__pos, tensor.et))
+  }
+
+  def Tanh(tensor: TENSOR, anno: Anno = NAnno)(implicit __pos: SourceContext): TENSOR = {
+    val res_tt = tensor.resultType
+    (new TENSOR(Adapter.g.reflectRead("tensor_tanh", C(res_tt), C(anno), tensor.x)(tensor.x)).withSrcType(__pos, tensor.et))
+  }
+
+  def Relu(tensor: TENSOR, anno: Anno = NAnno)(implicit __pos: SourceContext): TENSOR = {
+    val res_tt = tensor.resultType
+    (new TENSOR(Adapter.g.reflectRead("tensor_relu", C(res_tt), C(anno), tensor.x)(tensor.x)).withSrcType(__pos, tensor.et))
+  }
+
   override def mergable_dims(node: Node) = node match {
     case Node(s, "tensor_transpose", _, _) => List()
     case Node(s, "tensor_negate", _, _) => List()
+    case Node(s, "tensor_invert", _, _) => List()
+    case Node(s, "tensor_tanh", _, _) => List()
+    case Node(s, "tensor_relu", _, _) => List()
```

Comment:
maybe we can build a local variable called `unaryops`, and here, use the `if unaryops.contains(op)` instead

Reply:
done.